### PR TITLE
Re-implement machine-specs

### PR DESCRIPTION
**Pull request by skyreflectedinmirrors**
Re-implement machine-specs to contain documentation, units, etc.
Print using tabulate.

### CODE REVIEW

File: src/utils/specs.py [hunk 1]
```diff
@@ -52,129 +53,264 @@
...
+        'name': 'Max MCLK',
+        'unit': 'MHz'})
+    cur_sclk: str = field(default=None, metadata={'doc': 
+        '[RESERVED] The current engine (compute unit) clock rate of the accelerators/GPUs in the system. Unused.',
+        'name': 'Max SCLK',
+        'unit': 'MHz'})
+    cur_mclk: str = field(default=None, metadata={'doc': 
+        '[RESERVED] The current memory clock rate of the accelerators/GPUs in the system. Unused.',
+        'name': 'Max MCLK',
+        'unit': 'MHz'})
```
Comment (coleramos425) [inline src/utils/specs.py:287]:
```suggestion
    max_sclk: str = field(default=None, metadata={'doc': 
        'The maximum engine (compute-unit) clock rate of the accelerators/GPUs in the system.',
        'name': 'Max SCLK',
        'unit': 'MHz'})
    max_mclk: str = field(default=None, metadata={'doc': 
        'The maximum memory clock rate of the accelerators/GPUs in the system.',
        'name': 'Max MCLK',
        'unit': 'MHz'})
    cur_sclk: str = field(default=None, metadata={'doc': 
        '[RESERVED] The current engine (compute unit) clock rate of the accelerators/GPUs in the system. Unused.',
        'name': 'Cur SCLK',
        'unit': 'MHz'})
    cur_mclk: str = field(default=None, metadata={'doc': 
        '[RESERVED] The current memory clock rate of the accelerators/GPUs in the system. Unused.',
        'name': 'Cur MCLK',
        'unit': 'MHz'})
```
I think this is what you meant to say?

File: src/utils/specs.py [hunk 2]
```diff
@@ -52,129 +53,264 @@
...
+        'doc': 'The hardware blocks profiling information was collected for.',
+        'name': 'IP Blocks',
+        'optional': True
+        })
+    timestamp: str = field(default=None, metadata={
+        'doc': 'The time (in local system time) when data was collected',
+        'name': 'Timestamp'})
+    version: str = field(default=None, metadata={
+        'doc': 'The version of the machine specification file format.',
+        'name': 'MachineSpecs Version'})
```
Comment (coleramos425) [inline src/utils/specs.py:189]: OCD, but I'm not crazy about embedding version directly in the table. Maybe just printing it above like
image

I'll see if that can be changed

**Review by coleramos425 — CHANGES_REQUESTED** [review]

File: src/utils/specs.py (hunk 1 above)
Comment (skyreflectedinmirrors) [reply src/utils/specs.py:287]: Ah, yes -- thanks

File: src/utils/specs.py (hunk 2 above)
Comment (skyreflectedinmirrors) [reply src/utils/specs.py:189]: Ah, yeah -- that's doable. I can push a patch, 1 sec

Comment (skyreflectedinmirrors) [reply src/utils/specs.py:189]: image

Comment (skyreflectedinmirrors) [reply src/utils/specs.py:189]: Done in 0b566151dade65b63857467a93e661047342facc

**Comment by coleramos425** [comment]
LGTM. Thanks, Nick!

**Review by coleramos425 — APPROVED** [review]

**Pull request merged**